### ISSUE TITLE
Couple PHP 8 Errors

### ISSUE THREAD

**Issue by andrewdalpino** [zephir-lang/zephir#2165]
Hey there!

The Tensor 2.2 extension compiled just fine on PHP 8.0.3 with Zephir 0.3.0-beta-1, however we have a couple errors and failed tests now.

The `reciprocal()` methods (see https://github.com/RubixML/Tensor/blob/master/tensor/matrix.zep#L1603 and https://github.com/RubixML/Tensor/blob/master/tensor/vector.zep#L1075) throw

```
RuntimeException: Trying to call method dividematrix on a non-object
```

when trying to access the `this` variable in the `reciprocal()` method context. Here's the generated C code.

```c
/**
 * Return the reciprocal of the vector element-wise.
 *
 * @return self
 */
PHP_METHOD(Tensor_Vector, reciprocal) {

	zval _0, _1;
	zephir_method_globals *ZEPHIR_METHOD_GLOBALS_PTR = NULL;
	zend_long ZEPHIR_LAST_CALL_STATUS;
	zval *this_ptr = getThis();

	ZVAL_UNDEF(&_0);
	ZVAL_UNDEF(&_1);


	ZEPHIR_MM_GROW();

	zephir_read_property(&_1, this_ptr, ZEND_STRL("n"), PH_NOISY_CC | PH_READONLY);
	ZEPHIR_CALL_STATIC(&_0, "ones", NULL, 0, &_1);
	zephir_check_call_status();
	ZEPHIR_RETURN_CALL_METHOD(&_0, "dividevector", NULL, 0, this_ptr);
	zephir_check_call_status();
	RETURN_MM();

}
```

Also, this is the generated C code of a static method `zeros()` (see https://github.com/RubixML/Tensor/blob/master/tensor/vector.zep#L62) that is supposed to instantiate a Vector of length n filled with zeros

```c
/**
 * Build a vector of zeros with n elements.
 *
 * @param int n
 * @throws \Tensor\Exceptions\InvalidArgumentException
 * @return self
 */
PHP_METHOD(Tensor_Vector, zeros) {

	zephir_method_globals *ZEPHIR_METHOD_GLOBALS_PTR = NULL;
	zval *n_param = NULL, _0, _1;
	zend_long n, ZEPHIR_LAST_CALL_STATUS;
	zval *this_ptr = getThis();

	ZVAL_UNDEF(&_0);
	ZVAL_UNDEF(&_1);
#if PHP_VERSION_ID >= 80000
	bool is_null_true = 1;
	ZEND_PARSE_PARAMETERS_START(1, 1)
		Z_PARAM_LONG(n)
	ZEND_PARSE_PARAMETERS_END();

#endif


	ZEPHIR_MM_GROW();
	zephir_fetch_params(1, 1, 0, &n_param);

	n = zephir_get_intval(n_param);


	ZVAL_LONG(&_0, 0);
	ZVAL_LONG(&_1, n);
	ZEPHIR_RETURN_CALL_STATIC("fill", NULL, 0, &_0, &_1);
	zephir_check_call_status();
	RETURN_MM();

}
```

However, it's returning `null` instead of `\Tensor\Vector`. I found it suspicious that `ones()` (see https://github.com/RubixML/Tensor/blob/master/tensor/vector.zep#L74) worked fine however.

**Comment by Jeckerson** [comment]
Related #2167

**Comment by Jeckerson** [comment]
Fixed in #2166
